runtime/serde/serde-json/common/test/aws/smithy/kotlin/runtime/serde/json/JsonDeserializerTest.kt

```diff
@@ -362,6 +362,110 @@ class JsonDeserializerTest {
         assertTrue(found, "unknown field not enumerated")
     }
 
+    private class BasicUnionTest {
```

Comment:
These tests aren't really providing any additional value (especially since this behavior is covered by protocol tests anyway).